benchmarks/circuit_construction.py

```diff
@@ -0,0 +1,110 @@
+                z_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+            else:
+                x_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+    return data_qubits, z_measure_qubits, x_measure_qubits
+
+
+def rotated_surface_code_memory_z_cycle(
+    data_qubits, z_measure_qubits, x_measure_qubits, z_order, x_order
+) -> cirq.OP_TREE:
+    yield cirq.Moment([cirq.H(q) for q in x_measure_qubits])
+    for k in range(4):
+        op_list = []
+        for measure_qubits, add, is_x in [
+            [x_measure_qubits, x_order[k], True],
+            [z_measure_qubits, z_order[k], False],
+        ]:
+            for q_meas in measure_qubits:
+                q_data = q_meas + add
+                if q_data in data_qubits:
+                    op_list.append(cirq.CNOT(q_meas, q_data) if is_x else cirq.CNOT(q_data, q_meas))
+        yield cirq.Moment(op_list)
+    yield cirq.Moment([cirq.H(q) for q in x_measure_qubits])
+    yield cirq.Moment(cirq.measure_each(*x_measure_qubits, *z_measure_qubits))
+
+
+def surface_code_circuit(
+    distance: int, num_rounds: int, moment_by_moment: bool = True
+) -> cirq.Circuit:
+    data_qubits, z_measure_qubits, x_measure_qubits = get_qubits(distance)
+    x_order = [(1, 1), (1, -1), (-1, 1), (-1, -1)]
+    z_order = [(1, 1), (-1, 1), (1, -1), (-1, -1)]
+    surface_code_cycle = cirq.Circuit(
+        rotated_surface_code_memory_z_cycle(
+            data_qubits, x_measure_qubits, z_measure_qubits, x_order, z_order
+        )
+    )
+    if moment_by_moment:
+        return cirq.Circuit(
+            surface_code_cycle * num_rounds, cirq.Moment(cirq.measure_each(*data_qubits))
+        )
+    else:
+        return cirq.Circuit(
+            [*surface_code_cycle.all_operations()] * num_rounds, cirq.measure_each(*data_qubits)
+        )
+
+
+class SurfaceCode_Rotated_Memory_Z:
+    pretty_name = "Surface Code Rotated Memory-Z Benchmarks."
+    params = [*range(3, 26, 2)]
+    param_names = ["distance"]
+
+    def time_circuit_construction_moment_by_moment(self, distance: int) -> None:
+        """Benchmark circuit construction for Rotated Bottom-Z Surface code."""
+        _ = surface_code_circuit(distance, distance * distance)
+
+    def time_circuit_construction_operations_by_operation(self, distance: int) -> None:
+        """Benchmark circuit construction for Rotated Bottom-Z Surface code."""
+        _ = surface_code_circuit(distance, distance * distance, False)
+
+    def track_circuit_operation_count(self, distance: int) -> int:
+        """Benchmark operation count for Rotated Bottom-Z Surface code."""
+        circuit = surface_code_circuit(distance, distance * distance)
+        return sum(1 for _ in circuit.all_operations())
+
+    def track_circuit_depth(self, distance: int) -> int:
+        """Benchmark operation count for Rotated Bottom-Z Surface code."""
+        circuit = surface_code_circuit(distance, distance * distance)
+        return len(circuit)
+
+
+class NDCircuit:
```

Comment:
What is "ND" circuit?  Avoid abbreviation in naming:
go/pystyle#naming

Reply:
Updated.